kustomize/sds-agent/sds-agent.yaml

```diff
@@ -0,0 +1,128 @@
+---
+# Source: nodeagent/templates/serviceaccount.yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: istio-sdsagent
+  namespace: istio-system
+  labels:
+    app: istio-sdsagent
+    release: istio-system-istio-sdsagent
+
+---
+# Source: nodeagent/templates/clusterrole.yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: istio-sdsagent-istio-system
+  labels:
+    app: istio-sdsagent
+    release: istio-system-istio-sdsagent
+rules:
+- apiGroups: [""]
+  resources: ["configmaps"]
+  verbs: ["get"]
+
+---
+# Source: nodeagent/templates/clusterrolebinding.yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: istio-sdsagentnt-istio-system
+  labels:
+    app: istio-sdsagent
+    release: istio-system-istio-sdsagent
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: istio-sdsagent-istio-system
+subjects:
+  - kind: ServiceAccount
+    name: istio-sdsagent
+    namespace: istio-system
+
+---
+# Source: nodeagent/templates/daemonset.yaml
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: istio-sdsagent
+  namespace: istio-system
+  labels:
+    app: istio-sdsagent
+    istio: sdsagent
+    release: istio-system-istio-sdsagent
+spec:
+  selector:
+    matchLabels:
+      istio: sdsagent
+  template:
+    metadata:
+      labels:
+        app: istio-sdsagent
+        istio: sdsagent
+        release: istio-system-istio-sdsagent
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      serviceAccountName: istio-sdsagent
+      containers:
+        - name: sdsagent
+          image: "gcr.io/istio-release/node-agent-k8s:master-latest-daily"
+          imagePullPolicy: Always
+          args:
+            - --log_output_level=all:debug
+          volumeMounts:
+            - mountPath: /var/run/sds
+              name: sdsudspath
+          env:
+            - name: CA_PROVIDER
+              value: Citadel
+            - name: CA_ADDR
+              value: istio-citadel:8060
+            - name: "Trust_Domain"
```

Comment:
in istio/istio, trustdomain is global configmap/meshconfig and passed to by pilot/nodagent/citadel/etc, how new installer plan to handle trust domain ?

Reply:
New installer doesn't have any 'globals' ( to reduce risks ), by design.
And the meshconfig is never exposed to user - all is from either values.yaml or kustomizations.